### PR TITLE
ZIO 2.0.0-RC3

### PR DESCRIPTION
**Pull request by vigoo**

### CODE REVIEW

File: project/BuildHelper.scala [hunk 1]
```diff
@@ -43,7 +43,7 @@ object BuildHelper {
     sys.props.get(property).map(_.toBoolean).getOrElse(default)
 
   private def customOptions =
-    if (propertyFlag("fatal.warnings", true)) {
+    if (propertyFlag("fatal.warnings", false)) {
```
Comment (vigoo) [inline project/BuildHelper.scala:46]: `ZIO.scoped` creates type inferred to `Any` errors that I could not get rid of

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala [hunk 2]
```diff
@@ -289,37 +293,36 @@ object Adapters {
...
+    scoped[R, E, ZStream[R, E, A]](fa).flatten
 
-  private def managed[R, E, A](managed: => ZManaged[R, E, A])(implicit trace: ZTraceElement): ZStream[R, E, A] =
-    new ZStream(managedOut(managed.map(Chunk.single)))
+  private def scoped[R, E, A](zio: => ZIO[R with Scope, E, A])(implicit trace: ZTraceElement): ZStream[R, E, A] =
+    new ZStream(scopedOut[R, E, Chunk[A]](zio.map(Chunk.single(_))))
 
-  private def managedOut[R, E, A](
-    m: => ZManaged[R, E, A]
+  private def scopedOut[R, E, A](
```
Comment (vigoo) [inline src/main/scala/zio/interop/reactivestreams/Adapters.scala:309]: I'm not sure we need these custom `scoped` and `scopedOut` implementations anymore. And I'm also not sure I translated the old implementation correctly :)

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala [hunk 3]
```diff
@@ -35,13 +35,13 @@ object Adapters {
...
+  )(implicit trace: ZTraceElement): ZIO[Scope, Nothing, (E => UIO[Unit], ZSink[Any, Nothing, I, I, Unit])] = {
     val sub = subscriber
     for {
-      error       <- Promise.makeManaged[E, Nothing]
+      error       <- Promise.make[E, Nothing]
       subscription = new DemandTrackingSubscription(sub)
-      _           <- ZManaged.succeed(sub.onSubscribe(subscription))
-      fiber       <- error.await.catchAll(t => UIO(sub.onError(t))).toManaged.fork
+      _           <- ZIO.succeed(sub.onSubscribe(subscription))
+      fiber       <- error.await.catchAll(t => UIO.succeed(sub.onError(t))).fork
```
[the remark this answers is not in the document]
Comment (vigoo) [reply src/main/scala/zio/interop/reactivestreams/Adapters.scala:44]: Yes I think this should be fixed

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala [hunk 4]
```diff
@@ -54,27 +54,27 @@ object Adapters {
       for {
         subscriberP    <- makeSubscriber[O](bufferSize)
         (subscriber, p) = subscriberP
-        _              <- ZManaged.acquireReleaseSucceed(publisher.subscribe(subscriber))(subscriber.interrupt())
-        subQ           <- ZManaged.fromZIOUninterruptible(p.await.interruptible)
+        _              <- ZIO.acquireRelease(ZIO.succeed(publisher.subscribe(subscriber)))(_ => UIO.succeed(subscriber.interrupt()))
```
[the remark this answers is not in the document]
Comment (vigoo) [reply src/main/scala/zio/interop/reactivestreams/Adapters.scala:57]: I vote for `ZIO` :)

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala [hunk 5]
```diff
@@ -54,27 +54,27 @@ object Adapters {
...
+        subQ           <- p.await.interruptible
         (sub, q)        = subQ
         process        <- process(sub, q, () => subscriber.await(), () => subscriber.isDone)
       } yield process
-    val pull = pullOrFail.catchAll(e => ZManaged.succeed(Pull.fail(e)))
-    fromPull(pull)
+    val pull = pullOrFail.catchAll(e => ZIO.succeed(Pull.fail(e)))
+    fromPull[Any, Throwable, O](pull)
```
[the remark this answers is not in the document]
Comment (vigoo) [reply src/main/scala/zio/interop/reactivestreams/Adapters.scala:63]: I think this did not compile without the type annotation

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (vigoo) [reply src/main/scala/zio/interop/reactivestreams/Adapters.scala:44]: Checked it again, `subscriberToSink` is still scoped so I think there is nothing to fix here?

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala [hunk 6]
```diff
@@ -41,7 +41,7 @@ object Adapters {
       error       <- Promise.make[E, Nothing]
       subscription = new DemandTrackingSubscription(sub)
       _           <- ZIO.succeed(sub.onSubscribe(subscription))
-      fiber       <- error.await.catchAll(t => UIO.succeed(sub.onError(t))).fork
+      fiber       <- error.await.catchAll(t => UIO.succeed(sub.onError(t))).forkScoped
```
Comment (runtologist) [inline src/main/scala/zio/interop/reactivestreams/Adapters.scala]: `.forkScoped` my new favorite operator. :)

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (runtologist) [reply src/main/scala/zio/interop/reactivestreams/Adapters.scala:63]: :thinking:

**Review by runtologist — APPROVED** [review]
LGTM. Thank you @vigoo.

**Pull request merged**